graphql/statistics/fields.ts

```diff
@@ -216,89 +257,210 @@ export class StatisticsResolver {
+    @Authorized(Role.ADMIN)
+    async numPupilsOnWaitingList(@Root() statistics: Statistics) {
+        return await prisma.waiting_list_enrollment.count({
+            distinct: 'pupilId',
+            where: {
+                AND: [{ createdAt: { gte: statistics.from } }, { createdAt: { lt: statistics.to } }],
+            },
+        });
+    }
+
+    @FieldResolver(() => Int)
+    @Authorized(Role.ADMIN)
+    async numCourseAppointments(@Root() statistics: Statistics, @Arg('category', { nullable: true }) category?: course_category_enum) {
+        return await prisma.lecture.count({
+            where: {
+                appointmentType: 'group',
+                subcourse: category != null ? { course: { category: { equals: category } } } : undefined,
+                AND: [{ createdAt: { gte: statistics.from } }, { createdAt: { lt: statistics.to } }],
+            },
+        });
+    }
+
+    @FieldResolver(() => [Bucket])
+    @Authorized(Role.ADMIN)
+    async timeCommitment(@Root() statistics: Statistics) {
+        const matches = await prisma.match.findMany({
+            where: { AND: [{ createdAt: { gte: statistics.from } }, { createdAt: { lt: statistics.to } }] },
+            orderBy: { createdAt: 'asc' },
+        });
+        const intervals = new Map<number, number>();
+        for (const match of matches) {
+            let dateUntil;
+            if (!match.dissolved || match.dissolvedAt < match.createdAt) {
+                dateUntil = new Date();
+            } else {
+                dateUntil = match.dissolvedAt;
+            }
+            const matchDuration = dateUntil.getTime() - match.createdAt.getTime();
+
+            intervals.set(match.studentId, matchDuration + (intervals.get(match.studentId) ?? 0));
+        }
+        let buckets: Bucket[] = [
+            {
+                from: 0,
+                to: 14 * 24 * 3600 * 1000,
+                value: 0,
+                label: '0-14 Tage',
+            },
+            {
+                from: 14 * 24 * 3600 * 1000,
+                to: 28 * 24 * 3600 * 1000,
+                value: 0,
+                label: '14-28 Tage',
+            },
+            {
+                from: 28 * 24 * 3600 * 1000,
+                to: 60 * 24 * 3600 * 1000,
+                value: 0,
+                label: '28-60 Tage',
+            },
+            {
+                from: 60 * 24 * 3600 * 1000,
+                to: 120 * 24 * 3600 * 1000,
+                value: 0,
+                label: '60-120 Tage',
+            },
+            {
+                from: 120 * 24 * 3600 * 1000,
+                to: 300 * 24 * 3600 * 1000,
+                value: 0,
+                label: '120-300 Tage',
+            },
+            {
+                from: 300 * 24 * 3600 * 1000,
+                to: -1,
+                value: 0,
+                label: '300+ Tage',
+            },
+        ];
+        intervals.forEach((duration, key) => {
+            buckets = buckets.map((b) => {
```

Comment:
This will rebuild the whole array at each iteration, making it perform badly from a space and time complexity perspective (also as it returns undefined it'll clear all other buckets). 

Although functional programming has its merits, I'd prefer something like:

buckets.find(it => ...).value += 1;